File: GeneratedEntityReaderClasses/Components/SpriteAnimatorComponent.cs
namespace GeneratedNoitaClasses;
public class SpriteAnimatorComponent : NoitaComponentBase,  iNoitaType<SpriteAnimatorComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString target_sprite_comp_name { get; set; } = new();
	public NoitaBool rotate_to_surface_normal { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		target_sprite_comp_name.Read(s);
		rotate_to_surface_normal.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		target_sprite_comp_name.Write(s);
		rotate_to_surface_normal.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/SpriteAnimatorComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString

from NoitaStream import NoitaStream
@dataclass
class SpriteAnimatorComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    target_sprite_comp_name:NoitaString = field(default_factory=NoitaString)
    rotate_to_surface_normal:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.target_sprite_comp_name.Read(s)
        self.rotate_to_surface_normal.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.target_sprite_comp_name.Write(s)
        self.rotate_to_surface_normal.Write(s)

